### PR TITLE
feat(api_event_errors): error field in APIEvents

### PR DESCRIPTION
**Pull request by ivor-juspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Addition of error field to APIEvent struct

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Addition of error field to APIEvent struct to log output errors

## How did you test it?
run below API twice
2nd request APIEvent response should contain error field with below error
`"error\":{\"type\":\"BadRequest\",\"value\":{\"sub_code\":\"HE\",\"error_identifier\":1,\"error_message\":\"The merchant account with the specified details already exists in our records\",\"extra\":null}}`

`curl --location 'http://localhost:8080/accounts' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data-raw '{
  "merchant_id": "ivor_merchant",
  "locker_id": "m0010",
  "merchant_name": "NewAge Retailer",
  "merchant_details": {
    "primary_contact_person": "John Test",
    "primary_email": "JohnTest@test.com",
    "primary_phone": "sunt laborum",
    "secondary_contact_person": "John Test2",
    "secondary_email": "JohnTest2@test.com",
    "secondary_phone": "cillum do dolor id",
    "website": "www.example.com",
    "about_business": "Online Retail with a wide selection of organic products for North America",
    "address": {
      "line1": "1467",
      "line2": "Harrison Street",
      "line3": "Harrison Street",
      "city": "San Fransico",
      "state": "California",
      "zip": "94122",
      "country": "US"
    }
  },
  "return_url": "https://google.com/success",
  "webhook_details": {
    "webhook_version": "1.0.1",
    "webhook_username": "ekart_retail",
    "webhook_password": "password_ekart@123",
    "payment_created_enabled": true,
    "payment_succeeded_enabled": true,
    "payment_failed_enabled": true
  },
  "routing_algorithm": {
    "type": "single",
    "data": "stripe"
  },
  "sub_merchants_enabled": false,
  "metadata": {
    "city": "NY",
    "unit": "245"
  }
}'`

grafana link to test

https://grafana.staging.eu.juspay.net/explore?orgId=1&left=%7B%22datasource%22:%22x5WTfCG4k%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22bach%5C%22%7D%20%7C%3D%20%60ivor_merchant%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22x5WTfCG4k%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by ivor-juspay** [comment]
Screenshot 2023-11-17 at 12 06 22 AM

File: crates/router/src/analytics/core.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ pub async fn get_domain_info(domain: AnalyticsDomain) -> AnalyticsApiResponse<Ge
             dimensions: utils::get_refund_dimensions(),
         },
     };
-    Ok(ApplicationResponse::Json(info))
+    Err(AnalyticsError::UnknownError.into())
 }
```
Comment (lsampras) [inline crates/router/src/analytics/core.rs:33]: did u mean to commit this?

File: crates/router/src/services/api.rs [hunk 2]
```diff
@@ -854,7 +856,15 @@ where
...
             metrics::request::track_response_status_code(res)
         }
-        Err(err) => err.current_context().status_code().as_u16().into(),
+        Err(err) => {
+            error.replace(
+                masking::masked_serialize(&err.current_context())
+                    .into_report()
+                    .attach_printable("Failed to serialize json response")
+                    .change_context(errors::ApiErrorResponse::InternalServerError.switch())?,
+            );
```
Comment (lsampras) [inline crates/router/src/services/api.rs:865]: don't need masking serialize here,
just plain `serde_json::to_value` would work....

File: crates/router/src/analytics/core.rs (hunk 1 above)
Comment (ivor-juspay) [reply crates/router/src/analytics/core.rs]: reverted

**Review by lsampras — APPROVED** [review]
LGTM...

**Review by harsh-sharma-juspay — APPROVED** [review]

**Pull request merged**